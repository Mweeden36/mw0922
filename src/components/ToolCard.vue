<template>
  <div class="tool-card__wrapper" :class="code">
    <ToolCardImg :code="code" class="tool-card-image" />
    <div class="tool-card-info">
      <h1 class="tool-card-title">{{ brand }} {{ toolType }}</h1>
      <div class="font-size--16" v-if="dailyRate">
        Daily Rate: <b class="tool-rental-rate">${{ dailyRate }}</b>
      </div>
    </div>
  </div>
</template>

<script>
import ToolCardImg from "@/components/ToolCardImg.vue";
import rentalChargesMixin from "@/mixins/rentalCharges";

export default {
  props: {
    brand: {
      type: String,
      required: true,
    },

    code: {
      type: String,
      required: true,
    },

    toolType: {
      type: String,
      required: true,
    },
  },

  mixins: [rentalChargesMixin],

  components: {
    ToolCardImg,
  },
};
</script>

<style lang="scss" scoped>
@import "../styles/colors";

.tool-card__wrapper {
  display: flex;
  box-sizing: border-box;
  cursor: pointer;
  text-decoration: none;
  .tool-card-info {
    padding: 0px 5px;
    .tool-card-title {
      padding-bottom: 5px;
    }
  }
}
</style>
