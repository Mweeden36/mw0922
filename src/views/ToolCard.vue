<script setup>
import { computed } from 'vue';
import ToolCardImg from '@/views/ToolCardImg.vue';
import rentalCharges from '@/data/rental_charges';

const props = defineProps({
  brand: {
    type: String,
    required: true,
  },

  code: {
    type: String,
    required: true,
  },

  type: {
    type: String,
    required: true,
  },
});

const dailyRate = computed(() => {
  return rentalCharges.find(rentalCharge => rentalCharge.type === props.type).dailyCharge;
});
</script>


<template>
  <div class="tool-card__wrapper">
    <ToolCardImg :code="code" class="tool-card-image" />
    <div class="tool-card-info">
      <h1 class="tool-card-title">
        {{ brand }} {{ type }}
      </h1>
      <div class="font-size--16">
        Daily Rate: <b>${{ dailyRate }}</b>
      </div>
    </div>
  </div>
</template>

<style lang="scss" scoped>
@import '../styles/colors';

.tool-card__wrapper {
  display: flex;
  box-sizing: border-box;
  padding: 5px;
  margin: 10px 0px;
  cursor: pointer;
  &:hover {
    box-shadow: 1px 1px 2px 1px $color-tertiary;
  }
  .tool-card-info {
    padding: 0px 5px;
    .tool-card-title {
      padding-bottom: 5px;
    }
  }
}
</style>