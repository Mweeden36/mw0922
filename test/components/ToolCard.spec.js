import ToolCard from "../../src/components/ToolCard.vue";
import { describe, it, expect } from "vitest";
import { shallowMount } from "@vue/test-utils";

describe("ToolCard.vue", () => {
  it("should initialize with sensible defaults", () => {
    const { vm } = shallowMount(ToolCard, {
      propsData: {
        code: "JAKD",
        toolType: "jackhammer",
        brand: "DeWalt",
      },
    });
    expect(vm.code).toEqual("JAKD");
    expect(vm.toolType).toEqual("jackhammer");
    expect(vm.brand).toEqual("DeWalt");
    expect(vm.dailyRate).toEqual(2.99);
  });
});
