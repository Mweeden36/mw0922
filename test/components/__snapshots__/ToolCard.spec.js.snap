// Vitest Snapshot v1

exports[`ToolCard.vue > should generate the correct html 1`] = `
"<div class=\\"tool-card__wrapper JAKD\\" data-v-589eb99b=\\"\\">
  <tool-card-img-stub code=\\"JAKD\\" class=\\"tool-card-image\\" data-v-589eb99b=\\"\\"></tool-card-img-stub>
  <div class=\\"tool-card-info\\" data-v-589eb99b=\\"\\">
    <h1 class=\\"tool-card-title\\" data-v-589eb99b=\\"\\">DeWalt jackhammer</h1>
    <div class=\\"font-size--16\\" data-v-589eb99b=\\"\\"> Daily Rate: <b class=\\"tool-rental-rate\\" data-v-589eb99b=\\"\\">$2.99</b></div>
  </div>
</div>"
`;
